tests/conftest.py

```diff
@@ -170,13 +178,39 @@ def _generate_calculation_node(process_state=ProcessState.FINISHED, exit_status=
         if process_state is ProcessState.FINISHED and exit_status is None:
             exit_status = 0
 
-        node = CalculationNode(process_type=entry_point)
-        node.set_process_state(process_state)
+        calculation_node = CalculationNode(process_type=entry_point)
+        calculation_node.set_process_state(process_state)
 
         if exit_status is not None:
-            node.set_exit_status(exit_status)
+            calculation_node.set_exit_status(exit_status)
+
+        if repository is not None:
+            # Or use from_tree?
+            calculation_node.base.repository.put_object_from_filelike(repository[0], repository[1])
```

Comment:
I don't understand this. If `repository` is of type `pathlib.Path`, why are we indexing it? I would indeed just have `repository` take a `pathlib.Path` so keeping the type hint. The caller should then simply generate the directory structure as they see fit and here you call `put_object_from_tree` and that's it. Simplest and most flexible solution

Reply:
The reason for the indexing was to set the `BinaryIO` `handle` and `path` separately (in that case, the `Path` type hint was actually not appropriate when using `put_object_from_filelike`). For this test, I only want to add the file `file.txt` with `filecontent` "a" to the repository, so I don't actually care about any nested structure/tree here. It's just a flat file in the repository, but with a certain content. I'm not sure how I can construct a `pathlib.Path` object that has this structure? Using a `tempfile`? 
Looking at the `put_object_from_tree` signature:
```python
def put_object_from_tree(self, filepath: str, path: str | None = None):
```
It should be simple enough to store a file reference without content via:
```python
calculation_repository = Path('file.txt')
calculation_node.base.repository.put_object_from_tree(calculation_repository)
```
However, this throws an exception as `filepath` needs to be absolute. However, when using `resolve()` the resulting absolute path points to a non-existing file, so nothing is stored in the repository. Again, I could create this file as a `tempfile`, but then the `tempfile` has an actual path on the file system, while I also want to also specify the relative `path` in the repository. So I don't know how I can achieve what I want by passing a single `Path` argument to the `put_object_from_tree` method? I should require both, the `filepath` for the actual file, and the `path` for the relative path in the repo? Am I missing something?

Reply:
What I mean is that the caller, i.e. the test, should do the following in that case:
```python
def test(tmp_path, generate_calculation_node):
    repository = (tmp_path / 'file.txt').write_text('a')
    node = generate_calculation_node(..., repository=tmp_path)
```
When the fixture then just calls:
```python
calculation_node.base.repository.put_object_from_tree(repository)
```
The node's repo will contain a single file `file.txt` with the content `a`. That is what you are looking for, correct?

Reply:
Ah, OK, now I see. Yes, that works. Thanks!